service/src/geopm/CircularBuffer.hpp

```diff
@@ -176,22 +185,28 @@ namespace geopm
             throw Exception("CircularBuffer::insert(): Cannot insert into a buffer of 0 size", GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
         }
         if (m_count < m_max_size) {
-            m_buffer[m_count] = value;
+            m_buffer.at(m_count) = value;
             m_count++;
         }
         else {
-            m_buffer[m_head] = value;
+            m_buffer.at(m_head) = value;
             m_head = ((m_head + 1) % m_max_size);
         }
     }
 
     template <class type>
-    const type& CircularBuffer<type>::value(const unsigned int index) const
+    const type& CircularBuffer<type>::value(const int index) const
     {
-        if (index >= m_count) {
+        if (index >= 0 && index >= m_count ||
+            index <  0 && index < -m_count) {
```

Comment:
```suggestion
        if (index >= m_count || index < -m_count) {
```

`unsigned long m_count` cannot be negative (by data type and by nature of being a non-borrowable count), so `index >= m_count` implies `index >= 0` and `index < -m_count` implies `index < 0`.

Reply:
That change causes a lot of tests to fail, so it cannot be implemented.